modules/nf-google/src/main/nextflow/cloud/google/util/GsPathFactory.groovy

```diff
@@ -30,14 +33,29 @@ import nextflow.file.FileSystemPathFactory
 @CompileStatic
 class GsPathFactory extends FileSystemPathFactory {
 
+    private CloudStorageConfiguration storageConfig
+
+    GsPathFactory() {
+        try {
+            final config = GoogleLifeSciencesConfig.fromSession(Global.getSession() as nextflow.Session)
+            final builder = CloudStorageConfiguration.builder()
+            if (config.useRequesterPays) {
+                builder.userProject(config.project)
+            }
+            storageConfig = builder.build()
+        } catch(NullPointerException) {
+            storageConfig = CloudStorageConfiguration.DEFAULT
+        }
+    }
+
```

Comment:
Does this need to be done for each file? Could not the config created just once using [lazy initialization](https://github.com/nextflow-io/nextflow/blob/a5a6a1fc8f76a9a7ebbe98cc94b5356f2a23bdec/modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy#L47-L58)? 

Also the `catch(NullPointerException)` smell bad, what's the need for it?

Reply:
Changed in aa5550f

Reply:
Maybe it should an `IllegalStateException` if the session is missing.